torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+            self._extract(url)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+        else:
+            # Download the dataset
+            self._download(url, md5)
+            self._extract(url)
+
+    def _verify_split(self) -> None:
+        """Verify the integrity of the split file."""
+        # Download the dataset
+        download_url(
+            self.split_url,
+            self.root,
+            filename=os.path.split(self.split_url)[1],
+            md5=None,
+        )
+
+    def _download(self, url: str, md5: str) -> None:
+        """Download the dataset."""
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, url: str) -> None:
+        """Extract the dataset."""
+        filepath = os.path.join(self.root, os.path.split(url)[1])
+        extract_archive(filepath)
+
+    def _check_integrity(self, filename: str, md5: str) -> bool:
+        """Check integrity of dataset.
+
+        Returns:
+            True if dataset files are found and/or MD5s match, else False
+        """
+        filepath = os.path.join(self.root, os.path.split(filename)[1])
+        if not check_integrity(filepath, md5 if self.checksum else None):
+            return False
+        return True
+
+    def merge_parts(self, source_path: str, modality: str) -> None:
+        """Merge the modalities that are downloaded and extracted in two parts."""
+        fname_p1 = modality + "_part1"
+        fname_p2 = modality + "_part2"
+        source_folder = os.path.join(source_path, fname_p1)
+        destination_folder = os.path.join(source_path, fname_p2)
+        for file_name in os.listdir(source_folder):
+            source = os.path.join(source_folder, file_name)
+            destination = os.path.join(destination_folder, file_name)
+            if os.path.isfile(source):
+                shutil.move(source, destination)
+
+        shutil.rmtree(source_folder)
+        dest_split = os.path.split(destination_folder)
+        if len(modality.split("_")) == 3:
+            rename_dest = os.path.join(dest_split[0], "s2_temporal_subset")
+        if len(modality.split("_")) == 2:
+            rename_dest = os.path.join(
+                dest_split[0],
+                dest_split[1].split("_")[0] + "_" + dest_split[1].split("_")[1],
+            )
+        if "_" not in modality:
+            rename_dest = os.path.join(dest_split[0], dest_split[1].split("_")[0])
+        os.rename(destination_folder, rename_dest)
+
+    def convert_to_color(self, arr_2d: Any, palette: Any) -> Any:
```

Comment:
Any should be avoided unless the function is required to be dynamically typed. Let me know if you need any help with type hints